tasks/ModelTask.cpp

```diff
@@ -30,101 +29,102 @@ void ModelTask::setGazeboModel(WorldPtr _world,  ModelPtr _model)
-		ports()->addPort( *link_port );
-		link_port_list.push_back( std::make_pair(link_port,*link) );
+        for(NameVector::iterator it = link_names.begin(); it != link_names.end(); ++it)
+        {
+            if( (*it).compare( (*link)->GetName() ) == 0 )
+            {
+                gzmsg << "RockBridge: found link: " << world->GetName() + "/" + model->GetName() +
+                    "/" + (*link)->GetName() << std::endl;
+
+                // Create the ports dynamicaly
+                link_out_port = new RBSOutPort( "link_"+ (*link)->GetName() +"_samples" );
+                ports()->addPort(*link_out_port);
+                link_list.push_back( std::make_pair(link_out_port,*link) );
+            }
+        }
 	}
 }
-//======================================================================================
+
 void ModelTask::updateHook()
 {
     updateJoints();
     updateLinks();
 }
-//======================================================================================
+
 void ModelTask::updateJoints()
 {
-	for(JointPort_V::iterator it = joint_port_list.begin();it != joint_port_list.end(); ++it)
-	{
-		double effort = 0;	
-		(*it).first->readNewest(effort);
-		if(effort != RTT::NoData)
-		{
-			// Define the limits for the joint effort
-			try{
-			    if((-1.0 <= effort) && (effort <= 1.0)){
-			        (*it).second->SetForce(0,effort);
-			    }else{
-			        throw;
-			    }
-			}catch(...){
-			    // Joint effort value out of range 
-			    gzmsg << "RockBridge: error - effort value out of range (-1 <= effort <= 1)." << std::endl;
-			}
-		}
-	}
+    _joints_cmd.readNewest( joints_in );
+    for(Joint_V::iterator it = gazebo_joints.begin(); it != gazebo_joints.end(); ++it )
+    {
+        // Apply effort to joint
+        double effort = joints_in.getElementByName( (*it)->GetName() ).effort;
+        if( effort != RTT::NoData )
+            (*it)->SetForce(0,effort);
+
+        // Read joint angle from gazebo link
+        double angle = (*it)->GetAngle(0).Radian();
+        joints_out.getElementByName( (*it)->GetName() ).Position(angle);
+    }
+    _joints_samples.write( joints_out );
 }
-//======================================================================================
+
 void ModelTask::updateLinks()
 {
-	// Update all links from gazebo model
-	for(LinkPort_V::iterator it = link_port_list.begin(); it != link_port_list.end(); ++it)
-	{
-		LinkPtr link = (*it).second;
-
-		// Read rock input port and apply a force to gazebo links
-		base::Vector3d force(0.0, 0.0, 0.0);
-		(*it).first->readNewest(force);				
-		if((force(0) != RTT::NoData)||(force(1) != RTT::NoData)||(force(2) != RTT::NoData)){
-			link->AddRelativeForce( math::Vector3(force(0),force(1),force(2)) );
-		}
-	}
+    base::samples::RigidBodyState rock_rbs;
+    for(LinkOutput::iterator it = link_list.begin(); it != link_list.end(); ++it)
+    {
+        math::Pose link_pose = (*it).second->GetWorldPose();
```

Comment:
gazebo::physics::Entity::GetRelativePose() gets the pose of the entity relative to its parent.... 
So, to generalize the target I should take the difference from the source and target absolute positions, right?
How do you think we should define the source/targets ? Through properties ?  Instead of exported_links we use exported_sources/exported_targets ?